### ISSUE TITLE
Revised - Useless submerge subs prompt

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#6504]
### How can the problem be recreated?
- Map: WWII Revised
- set up a combat of a sub vs a transport
- fight a round of combat where both the subs and transport miss
- notice: there is a retreat or submerge prompt, click submerge
- notice: there is a retreat or remain prompt

### What is an expected fix?

Clicking 'retreat' sends the sub back to the territory from whence it came, the submerge prompt seems to have no effect and probably should not be displayed in such cases.

### (Optional) Additional information

This is an example scenario set up for this situation:
![Screenshot from 2020-05-20 19-03-40](https://user-images.githubusercontent.com/12397753/82515275-b6da7f00-9acc-11ea-921c-6f47af4294a1.png)

[test-revised-vs-solo-transport.tsvg.zip](https://github.com/triplea-game/triplea/files/4660239/test-revised-vs-solo-transport.tsvg.zip)

Save game is captured @ SHA: edafc921297149a7ace2f41a1f8b15694a9c2a25 (May 20, 2020)

**Comment by DanVanAtta** [comment]
@panther2 would you mind having a look at this scenario, could you confirm that indeed a submerge prompt is incorrect and there should only be a retreat prompt?

**Comment by panther2** [comment]
@DanVanAtta I have recreated the battle using a newer pre-release - and I get these windows after the dice have been rolled for both sides for the first time:
![1_Retreat_by_submerge](https://user-images.githubusercontent.com/20812775/82579895-bb676c00-9b8e-11ea-8407-020fe88418a3.PNG)
and klicking "Remain" I get:
![2_Retreat_by Retreat](https://user-images.githubusercontent.com/20812775/82579950-cfab6900-9b8e-11ea-9731-5d28b64bd8b2.PNG)

**Comment by DanVanAtta** [comment]
Does clicking 'remain' actually lead the sub to submerging?
I honestly had not considered the game engine would do that, 'Remain' typically means "remain in combat" in every other context. It seems that the "remain" option would actually be better worded as the proper submerge option.

It sounds like we probably should have one dialog with four options:
- Submerge
- Retreat
- Remain
- Ask me Later

In this way it is clear submerge ends the battle and remain retains the meaning it does in other battle prompts (stays consistent). WDYT @panther2?

**Comment by Cernelius** [comment]
> It sounds like we probably should have one dialog with four options:
> 
> Submerge
> Retreat
> Remain
> Ask me Later

Four options, if you have 1 submarine on the offensive. Otherwise, a number of options equal to 3 plus the number of submarines on the offensive, since you can submerge submarines individually (not sure if this applies to every ruleset) (of course, no submarine ever both submerges and retreats).

**Comment by Cernelius** [comment]
Also, even with an offensive force of 2 or more units of which (only) 1 is a submarine, what I understand is your full set of options, as the offender, during the "retreat" step, is:
1. Remain in the battle with everything.
2. Submerge the submarine and remain in the battle with everything else.
3. Retreat everything.
4. Submerge the submarine and retreat everything else.
5. To which one would add the "Ask me later" utility option.

**Comment by panther2** [comment]
@DanVanAtta 
>Does clicking 'remain' actually lead the sub to submerging?

Klicking "Submerge" in the first dialogue leads to submerging. Klicking "Remain" leads to the second dialogue.

I second @Cernelius postings.

**Comment by DanVanAtta** [comment]
What I see, first dialog:
- clicking remain keeps you in the battle
- clicking submerge opens the second dialog

On the second dialog if you did not click 'remain':
- clicking retreat, retreats
- clicking remain, submerges

The problem is that "remain" changes meaning from the first dialog to the second, it's not obvious that clicking 'submerge' and then 'remain' is the way to actually submerge.

**Comment by Cernelius** [comment]
Instead of the first prompt, you should have the ability to select all units you want to submerge (in this case, from 0 to X submarines, where X is the total number of not yet submerged submarines, but there may be games in which you have different unit types able to submerge). Then, the second prompt should be unrelated to what you just submerged, since submerged units stay in the zone of the battle while anything else retreats.

**Comment by Cernelius** [comment]
And, of course, you can submerge units over several battle rounds, whereas retreat happens only once (though it is not necessarily the end of the battle, therefore you can still submerge in a battle round after the battle round in which you retreated (talking about land units able to submerge in partially sea borne land battles)).

**Comment by DanVanAtta** [comment]
Yeah, after clicking submerge we should ask how many, then if any left show
the retreat dialog. Seems to make more sense

**Comment by Cernelius** [comment]
It should work like selecting casualties, with 0 being the default for each type. Conceptually, submerging is most similar to casualties selection, as in both cases you are removing units from the battle, selectively.

**Comment by panther2** [comment]
#4821

**Comment by trevan** [comment]
I've been looking at the retreat logic as part of the steps conversion.  I can fix the issue where it retreats the sub after it has been submerged.  And I can also fix the logic where it asks you to retreat even when you only have submerged subs.

**Comment by trevan** [comment]
#7007 prevents the second dialog that asks you to retreat after you've submerged all of the units.  This is a by-product of fixing the second issue mentioned in #4821

**Comment by DanVanAtta** [comment]
This update will help: https://github.com/triplea-game/triplea/pull/14378, renames buttons to reduce the confusion.

Remaining on this issue: the ability to submerge individual subs and not have it be all or nothing.
